src/CallbackSerializer.php

```diff
@@ -0,0 +1,52 @@
+<?php
+/**
+ * This file is part of the prooph/pdo-snapshot-store.
+ * (c) 2016-2017 prooph software GmbH <contact@prooph.de>
+ * (c) 2016-2017 Sascha-Oliver Prolic <saschaprolic@googlemail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Prooph\SnapshotStore;
+
+final class CallbackSerializer implements Serializer
+{
+    /**
+     * callable
+     */
+    private $serializeCallback = 'serialize';
+
+    /**
+     * callable
+     */
+    private $unserializeCallback = 'unserialize';
+
+    public function __construct(?callable $serializeCallback, ?callable $unserializeCallback)
+    {
+        if (null !== $serializeCallback && null !== $unserializeCallback) {
```

Comment:
don't you think we should discourage such behaviour?

Changing anything on the aggregate should not be done at the snapshot layer... When a snapshot does not exists or isn't up to date events are going to reconstitute it.

This check does not makes it *impossible* but atleast one might be more aware of that. What goes in comes out. Unless i'm missing something?